### PR TITLE
Sync to Slim\Http in 4.x 

### PR DESCRIPTION
**Pull request by akrabat**
Pick up all the changes to Slim\Http in Slim 4 by copying the code from Slim 4.x into this repository.

Notes:

* I've kept the `createFromEnvironment` to `createFromGlobals` rename
* I've deleted the `FactoryDefault` factory class.
* I've added PHPCS.
* I've added PHP 5.6 testing to Travis.
* I've removed completely base path - we need to handle this in Slim, not Slim-Http

### CODE REVIEW

File: src/HeadersInterface.php [hunk 1]
```diff
@@ -12,7 +12,7 @@
  * Headers Interface
  *
  * @package Slim
- * @since   3.0.0
+ * @since   1.0.0
```
Comment (silentworks) [inline src/HeadersInterface.php]: Should this not be `3.0.0` as it was before? we only got `HeadersInterface` in Slim 3?

Comment (akrabat) [inline src/HeadersInterface.php]: My view was that this will be since 1.0 of Slim-Http. 

Arguably the package should be Slim-Http though?

**Comment by geggleto** [comment]
In theory this looks pretty good.

File: composer.json [hunk 2]
```diff
@@ -23,18 +23,23 @@
...
-    "php": ">=7.0.0",
+    "php": ">=5.6",
     "psr/http-message": "^1.0"
   },
   "require-dev": {
+    "squizlabs/php_codesniffer": "^2.5",
     "phpunit/phpunit": "^5.7|^6.0"
   },
   "conflict": {
     "slim/slim": "^3.0"
```
Comment (geggleto) [inline composer.json]: Should this not be slim/slim ^4.0 as we will have namespacing conflicts?

Comment (akrabat) [inline composer.json]: Nope. Slim 4 won't have any of the `Slim\Http` namespace in it.

Comment (geggleto) [inline composer.json]: but say I include slim/http in a Slim 3 project ... ?

Comment (geggleto) [inline composer.json]: Oh my bad... it says conflict xD

File: composer.json [hunk 3]
```diff
@@ -23,18 +23,23 @@
...
   },
   "autoload": {
     "psr-4": {
-      "Slim\\Http\\": "src",
+      "Slim\\Http\\": "src"
```
Comment (silentworks) [inline composer.json]: I think from composer docs it says the path should have a / after it, so it should be `"src/"`

**Pull request merged**